dm/syncer/dml.go

```diff
@@ -108,7 +117,11 @@ RowLoop:
 			}
 		}
 
-		dmls = append(dmls, newDML(insert, param.safeMode, param.targetTableID, param.sourceTable, nil, value, nil, originalValue, columns, ti, nil))
+		if downstreamIndexColumns == nil {
+			downstreamIndexColumns = s.schemaTracker.GetAvailableDownStreamUKIndexInfo(tableID, value)
```

Comment:
Just asking, when will this function get a non-nil value while `downstreamIndexColumns == nil`? 🤔

Reply:
When the table has no `AbsoluteUKIndexInfo` , `downstreamIndexColumns == nil`, like no `pk/uk not null` , At L120.
When the table has no `uk` whoes value not null in this row, `downstreamIndexColumns` still is `nil`, At L121.